src/qr.jl

```diff
@@ -538,8 +538,10 @@ end
 function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
     m = min(size(A)...)
     m == 0 && return 0
-    tol = max(atol, rtol*abs(A.R[1,1]))
-    return something(findfirst(i -> abs(A.R[i,i]) <= tol, 1:m), m+1) - 1
+    rdiag = diag(getfield(A, :factors))
+    tol = max(atol, rtol*abs(rdiag[1]))
```

Comment:
`diag` makes an unnecessary copy:
```suggestion
    tol = max(atol, rtol*abs(getfield(A, :factors)[1,1]))
```